### DISCUSSION TITLE
Support needed: uat_debug:114

### DISCUSSION THREAD

**Discussion by fluxflax** [thomase1234/esphome-fake-xemex-csmb#23] [General]
Dear all,
I have set everything up as describes with an RS485 MAX module, PINs 16, 17, 18, 19 assigned accordingly.
However, the shell wallbox states a "warnings":["PowerMeterFailure"]

At the same time, the esp32 log shows ca every 3 seconds an uat_debug:114 message
At 11:03:22 I have manually adjusted the state value for CT1.

Does someone have any hint for me on how to find the reason for the warning/error?
Thank you very much!!

Here an example of the log:
11:01:12 | [D] | [uart_debug:114] | <<< 00:7F:FE:D9:FE:FD:6E:00
11:01:16 | [D] | [uart_debug:114] | <<< 00:7F:FE:D9:FE:FD:6E:00
11:03:22 | [D] | [number:054] | 'ct1current' - Setting number value
11:03:22 | [D] | [number:113] | New number value: 7.000000
11:03:22 | [D] | [modbus-server-xemex:310] | Current set to: 7.000000
11:03:22 | [D] | [modbus-server-xemex:322] | Registers for CT1 set to: 0x40e0 and 0x0000
11:03:22 | [D] | [number:012] | 'ct1current': Sending state 7.000000
11:03:36 | [D] | [uart_debug:114] | <<< 00:7F:FE:D9:FE:FD:6E:00
11:03:40 | [D] | [uart_debug:114] | <<< 00:7F:FE:D9:FE:FD:6E:00
11:03:44 | [D] | [uart_debug:114] | <<< 00:7F:FE:D9:FE:FD:6E:00
11:03:47 | [D] | [uart_debug:114] | <<< 00:7F:FE:D9:FE:FD:6E:00

**Comment by thomase1234** [comment]
Hi @fluxflax ,

Is this the same Shell charger device that is sending those commands? 
They don't look like Modbus RTU to me. 

You could try to powercycle the Shell charger and verify that the initial commands are similar to the ones I had in my logfile.
[log output](https://github.com/thomase1234/esphome-fake-xemex-csmb/blob/master/sample_output/serial_output.log)

`[11:27:12][D][uart_debug:114]: <<< 01:03:40:02:00:01:30:0A`
`[11:27:12][D][uart_debug:114]: >>> 01:03:02:51:42:05:E5`

**Reply by fluxflax** [reply]
Hi @thomase1234,
thank you for your support - I have exchanged the RS485 module and now it looks much better. With the adjusted setup the wallbox warning disappeared, and the logs look better. However, in each cycle (second) I have a warning" Component modbus_server took a long time for an operation (68ms). Is this important, or can I just ignore this?

![Wallbox](https://github.com/user-attachments/assets/281eca64-a81f-42f5-af57-dd7c3a3b242d)

What is even more important: even if I set the current ampere value to 32A (what should be the maximum), the wallbox does not reduce charging speed. Do you think, this is something shell has to solve or do you have any other idea, what I could check on my end?
Thank you very much
Florian

**Reply by thomase1234** [reply]
Hi Florian,

I think you can ignore the warning about the time it took to perform the operation. I've had those as well on my setup, and I don't see an impact.

Regarding reducing the charging speed. 
1. First of all, my wallbox was also misconfigured intially, so that indeed made it hard to diagnose what's wrong. The positive thing is that shell can fix that remotely.
2. You can try to figure out how your wallbox is configured by playing with the CT values. Do you have a 3 phase connection ? What happens when you put all 3 CT values to 40A ? If you only have a 40A single phase connection, try setting CT1 to 60 A and see if that stops the charging. If it stopped, you can lower the CT with 1A at a time and keep an eye on the charging speed. Once it starts charging, you should have an idea of what the max A is that has been configured on your wallbox. 
Suppose charging restarts at 30A, that would mean the Max is configured around 38A because the car needs at least 8A to start charging.

Good luck reverse engineering the configuration of your wallbox.

Best regards,
Thomas

**Reply by fluxflax** [reply]
Hi @thomase1234,
finally I managed to contact someone from Shell who was able to fix the configuration - the solution is now working, thank you again!!
I have still some quick questions:

- I asked Shell to set 40A on 3 phases as overall maximum for my house. As soon as I set 29A on each phase as current value, the wallbox states "9 Amp 3 phase" as value for "Limited now" and slows down charging accordingly. Does this mean, that Shell has set the maximum value for my house to 38A instead of 40A? Or do you know, whether the wallbox includes a buffer of 2A?
- Is it a problem, if the power meter sends a value of 39A in this case?
- In the web interface, the value for for "Max Limit" in the Status Section as well as the "Max Plug" in the connectors section is displayed as "32 Amp 3 phase", which is neither my overall limit (40A) nor the limit for the charger (16A). Do you know, what the meaning of these parameters is?

Best regards, Florian
